### PR TITLE
chore: add isVisible in default values in wds widgets

### LINKED ISSUE #35272: [Bug]: Incorrect visibility setting is toggled on the property pane for some widgets
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Incorrect visibility setting is toggled on the property pane for the following widgets:
 
 - table
 - input
 - currency
 - checkbox
 - radio group
 - switch
 - paragraph
 - header

image
image

### Steps To Reproduce

1. Drop above widgets and notice their visibility toggle value

### Public Sample App

_No response_

### Environment

Production

### Severity

High (Blocker to building or releasing)

### Issue video log

_No response_

### Version

cloud

### LINKED ISSUE #35317: [Bug]: Making parent widget invisible does not make child widget invisible
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Setting a section/zone visibility to false does not affect it's child widget's visibility. Only explicitly turning off visibility of a child widget alters its state.

### Steps To Reproduce

1. Drop a table widget on canvas as seen in the video
2. Turn off visibility of it's parent section/zone
3. Go to preview mode and observe the table is still seen
4. Turn off visibility of table widget
5. Go to preview mode and observe the table is not visible

### Public Sample App

_No response_

### Environment

Production

### Severity

Critical (Broken Production apps)

### Issue video log

https://www.loom.com/share/a90a5cf7419242828aaed074a3eb5089

### Version

cloud

### LINKED ISSUE #35319: [Bug]: Zone visiblilty property shows up as undefined because of which zone is always visible even when set to be invisible
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Zone visiblilty property shows up as undefined because of which zone is always visible even when set to be invisible. Attached video is for another [issue](https://github.com/appsmithorg/appsmith/issues/35317) but highlighted issue here can be observed from the 0.43s mark

### Steps To Reproduce

1. Drop any widget on canvas and set it's parent section visibility to true, zone to false and widget to true (make sure visual separation is on for the zone)
2. Switch to preview mode
3. Observe the zone show up

### Public Sample App

_No response_

### Environment

Production

### Severity

High (Blocker to building or releasing)

### Issue video log

https://www.loom.com/share/a90a5cf7419242828aaed074a3eb5089

### Version

cloud

### PR DESCRIPTION
**Pull request by jsartisan**
Fixes #35317
Fixes #35319
Fixes #35272

/ok-to-test tags="@tag.Anvil"

## Summary by CodeRabbit

- **New Features**
  - Introduced `autocompleteConfig` to manage visibility of autocomplete features consistently across the application.
  - Added `isVisible` property to `defaultConfig` for better control over widget visibility.

- **Improvements**
  - Simplified logic in the `shouldUpdateProperty` method for enhanced clarity and efficiency.
  - Updated `getAutocompleteDefinitions` methods to utilize the new `autocompleteConfig`, streamlining autocomplete handling.

- **Modifications**
  - Expanded export capabilities of configuration modules, including the new `autocompleteConfig` across multiple widgets.

> [!CAUTION]
> 🔴 🔴 🔴 Some tests have failed.
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10368473517>
> Commit: 362f2af4cd0370aefe57c0004102d25c9b10ad33
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10368473517&attempt=1&selectiontype=test&testsstatus=failed&specsstatus=fail" target="_blank">Cypress dashboard</a>.
> Tags: @tag.Anvil
> Spec: 
> The following are new failures, please fix them before merging the PR: <ol>
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilButtonWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilCheckboxGroupWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilCheckboxWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilCurrencyInputWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilHeadingWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilIconButtonWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilInlineButtonWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilInputWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilParagraphWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilPhoneInputWidgetSnapshot_spec.ts</ol>
> <a href="https://internal.appsmith.com/app/cypress-dashboard/identified-flaky-tests-65890b3c81d7400d08fa9ee3?branch=master" target="_blank">List of identified flaky tests</a>.
> <hr>Tue, 13 Aug 2024 11:27:23 UTC

### CODE REVIEW

File: app/client/src/components/propertyControls/BaseControl.tsx [hunk 1]
```diff
@@ -16,12 +16,14 @@ export type ControlMethods = Record<
...
 // eslint-disable-next-line @typescript-eslint/ban-types
 class BaseControl<P extends ControlProps, S = {}> extends Component<P, S> {
-  shoudUpdateProperty(propertyValue: unknown) {
-    return !(
-      (this.props.propertyValue === undefined &&
-        propertyValue === this.props.defaultValue) ||
-      !(this.props.propertyValue !== propertyValue)
-    );
+  shoudUpdateProperty(newValue: unknown) {
+    const { defaultValue, propertyValue: oldValue } = this.props;
```
Comment (jsartisan) [inline app/client/src/components/propertyControls/BaseControl.tsx:20]: This is a tiny refactor to make the code more readable. I just couldn't understand what was going on before.

File: app/client/src/widgets/anvil/SectionWidget/widget/config/autocompleteConfig.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { DefaultAutocompleteDefinitions } from "widgets/WidgetUtils";
+
+export const autocompleteConfig = {
+  isVisible: DefaultAutocompleteDefinitions.isVisible,
```
Comment (jsartisan) [inline app/client/src/widgets/anvil/SectionWidget/widget/config/autocompleteConfig.ts:4]: extracted the autocomplete config to separate file.

File: app/client/src/widgets/anvil/ZoneWidget/widget/config/autocompleteConfig.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { DefaultAutocompleteDefinitions } from "widgets/WidgetUtils";
+
+export const autocompleteConfig = {
+  isVisible: DefaultAutocompleteDefinitions.isVisible,
```
Comment (jsartisan) [inline app/client/src/widgets/anvil/ZoneWidget/widget/config/autocompleteConfig.ts:4]: extracted the autocomplete config to separate file.

File: app/client/src/components/propertyControls/BaseControl.tsx [hunk 4]
```diff
@@ -16,12 +16,14 @@ export type ControlMethods = Record<
...
-  shoudUpdateProperty(propertyValue: unknown) {
-    return !(
-      (this.props.propertyValue === undefined &&
-        propertyValue === this.props.defaultValue) ||
-      !(this.props.propertyValue !== propertyValue)
-    );
+  shouldUpdateProperty(newValue: unknown) {
+    const { defaultValue, propertyValue: oldValue } = this.props;
+
+    if (oldValue === undefined && newValue === defaultValue) return false;
```
Comment (sagar-qa007) [inline app/client/src/components/propertyControls/BaseControl.tsx:22]: Please add/update unit tests related to this function.

**Review by KelvinOm — APPROVED** [review]

**Pull request merged**